workflows_core/dataset/dataset.py

```diff
@@ -108,6 +117,53 @@ def get_documents(
         res["documents"] = [document.Document(d) for d in res["documents"]]
         return res
 
+    def get_all_documents(
+        self,
+        page_size: int = 64,
+        filters: Optional[List[Filter]] = None,
+        select_fields: Optional[List[str]] = None,
+        sort: Optional[list] = None,
+        include_vector: bool = True,
+        random_state: int = 0,
+        is_random: bool = False,
+        after_id: Optional[List] = None,
+        worker_number: int = 0,
+        max_retries: int = 3,
+    ) -> Dict[str, Any]:
+
+        documents = []
+        retry_count = 0
+        while True:
+            try:
+                chunk = self.get_documents(
+                    page_size=page_size,
+                    filters=filters,
+                    select_fields=select_fields,
+                    after_id=after_id,
+                    worker_number=worker_number,
+                    sort=sort,
+                    include_vector=include_vector,
+                    random_state=random_state,
+                    is_random=is_random,
+                )
+            except ConnectionError as e:
```

Comment:
For re-try logic, please add another error exception:  
- JSONDecodeErrors which support cloud-related errors that are usually a HTMLResponse